src/lib/Components/StickyTabPage/StickyTabPage.tsx

```diff
@@ -71,58 +69,31 @@ export const StickyTabPage: React.FC<{
           transform: [{ translateY: headerOffsetY as any }],
         }}
       >
-        <View onLayout={e => setHeaderHeight(e.nativeEvent.layout.height)}>
+        <View
+          onLayout={e => {
+            if (headerHeight) {
+              headerHeight.setValue(e.nativeEvent.layout.height)
+            } else {
+              setHeaderHeightNativeWrapper(new Animated.Value(e.nativeEvent.layout.height))
+            }
+          }}
+        >
           {headerContent}
           <Spacer mb={1} />
         </View>
-        <StickyTabBar>
-          {tabs.map(({ title }, index) => (
-            <StickyTab
-              key={title}
-              label={title}
-              active={activeTabIndex === index}
-              onPress={() => handleTabPress(index)}
-            />
-          ))}
-        </StickyTabBar>
+        <StickyTabPageTabBar
+          labels={tabs.map(({ title }) => title)}
+          initialActiveIndex={initialTabIndex}
+          onIndexChange={index => {
+            activeTabIndex.setValue(index)
+            railRef.current.setOffset(index * width)
+            tracking.trackEvent({
+              action_name: tabs[index].title,
+              action_type: Schema.ActionTypes.Tap,
+            })
+          }}
+        />
```

Comment:
Moved the tab bar into a separate file and reduced the coupling with this file 💆